sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -363,6 +370,15 @@ private[client] class Shim_v0_12 extends Shim with Logging {
     conf.getIntVar(HiveConf.ConfVars.METASTORE_CLIENT_CONNECT_RETRY_DELAY) * 1000L
   }
 
+  override def getTablesByType(
+      hive: Hive,
+      dbName: String,
+      pattern: String,
+      tableType: TableType): Seq[String] = {
+    throw new UnsupportedOperationException("Hive 2.2 and lower versions don't support " +
+      "getTablesByType. Please use Hive 2.3 or higher version.")
```

Comment:
@Eric5553, do you mean we'll support `SHOW VIEWS` only with Hive 2.3 in Spark side?

Reply:
Okay, so it's still possible to look up via `listTables`. Can you fix the PR description? seems not it's a precondition but just for better performance

Reply:
As @cloud-fan suggested in https://github.com/apache/spark/pull/27897#discussion_r394391663, for versions before Hive 2.3, I'll implement the fallback solution using `listTables` and filter the result in PR https://github.com/apache/spark/pull/27897.
Thanks for the review! @HyukjinKwon

Reply:
Sure, that's a mismatch. Updated the description, thanks :-)

Reply:
Can't we do the fallback here? It's better to make the low-level API completed, e.g. `getTablesByType` always sucess, but may be optimized with Hive 2.3+

Reply:
I thought `HiveShim` is a thin layer close to Hive API. Yea, it should be better if we handle it in low-level API. I'll make the change here.

Reply:
The fallback logic needs to call `HiveClient.getTablesByName`, which got a lot of logic in `HiveClient`. Thus I think it needs to handle the fallback logic in `HiveClient` instead of `HiveShim`. WDYT @maropu @cloud-fan @HyukjinKwon , thanks!

Also I renamed the `HiveClient.listViews` API to `HiveClient.listTablesByType` to provide a more common interface.

Reply:
We had better keep this `HiveShim` layer simple. +1 for @Eric5553 's suggestion to implement the fallback in `HiveClient` layer.

cc @gatorsmile